tinygrad/opsgpu.py

```diff
@@ -95,17 +95,16 @@ def binary_op(ctx, code, x, y):
   xdiv = 1
   ydiv = 1
   if x.shape != y.shape:
-    # special case broadcasting
-    # TODO: make general
-    if len(y.shape) == 4 and x.shape[0:2] == y.shape[0:2] and y.shape[2] == 1 and y.shape[3] == 1:
-      ydiv = x.shape[2] * x.shape[3]
-    elif len(y.shape) == 4 and x.shape[0:2] == y.shape[0:2] and x.shape[2] == 1 and x.shape[3] == 1:
-      xdiv = y.shape[2] * y.shape[3]
-    elif np.prod(y.shape) == 1:
-      ydiv = np.prod(x.shape)
-    else:
-      raise Exception("binary op shape mismatch: %r != %r" % (x.shape, y.shape))
-  ret = buffer_like(ctx, x if np.prod(x.shape) >= np.prod(y.shape) else y)
+    for axis, mismatch in enumerate(np.where(x.shape != y.shape, x.shape, y.shape)):
```

Comment:
Yes, you are right. I should have done way more testing, sorry :/ 
`np.where(x != y, 1, 0)` would actually be correct, otherwise the shapes are always compared as tuples. This can be shortened to `x != y` then.

Reply:
Yeah, I think you need to cast the shape tuples to arrays first either way. The shortest version is `np.array(x.shape) != np.array(y.shape), I believe.

Reply:
Yeah, just noticed that, as well :)